### PR TITLE
Switch development/tools to go mods

### PR DESCRIPTION
**Pull request by Ressetkk**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

#2268

### CODE REVIEW

File: development/tools/jobs/tester/componentsuite.go [hunk 1]
```diff
@@ -57,13 +57,13 @@ func (s ComponentSuite) Run(t *testing.T) {
 func (s ComponentSuite) preMasterTest(jobConfig config.JobConfig) func(t *testing.T) {
 	return func(t *testing.T) {
 		job := FindPresubmitJobByNameAndBranch(
-			jobConfig.Presubmits[s.repositorySectionKey()],
+			jobConfig.PresubmitsStatic[s.repositorySectionKey()],
```
Comment (tehcyx) [inline development/tools/jobs/tester/componentsuite.go:60]: I don't know why we have to change the naming as part of the go mod switch. Unless I'm missing something. Same for `Postsubmits` -> `PostsubmitsStatic`

File: development/tools/Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 resolve:
-	dep ensure -v -vendor-only
+	go mod verify
```
Comment (tehcyx) [inline development/tools/Makefile:2]: Can we change the Makefile to be in general more like e.g. this one: https://github.com/kyma-incubator/hydroform/blob/master/Makefile

File: prow/images/prow-tools/Makefile [hunk 3]
```diff
@@ -3,7 +3,7 @@ IMG = $(DOCKER_PUSH_REPOSITORY)$(DOCKER_PUSH_DIRECTORY)/$(IMG_NAME)
 TAG = $(DOCKER_TAG)
 
 resolve:
-	dep ensure -v -vendor-only
+	go mod verify
```
Comment (tehcyx) [inline prow/images/prow-tools/Makefile:6]: Same with this Makefile, as mentioned in the other.

File: development/tools/jobs/tester/componentsuite.go (hunk 1 above)
Comment (tehcyx) [reply development/tools/jobs/tester/componentsuite.go:60]: Nvm, I see why the change is necessary.

File: development/tools/jobs/testinfra/validate_test.go [hunk 4]
```diff
@@ -13,16 +13,16 @@ func TestValidateProwPresubmit(t *testing.T) {
 	jobConfig, err := tester.ReadJobConfig("./../../../../prow/jobs/test-infra/validation.yaml")
 	// THEN
 	require.NoError(t, err)
-	testInfraPresubmits := jobConfig.Presubmits["kyma-project/test-infra"]
+	testInfraPresubmits := jobConfig.AllStaticPresubmits([]string{"kyma-project/test-infra"})
```
Comment (tehcyx) [inline development/tools/jobs/testinfra/validate_test.go:16]: Here you already use the `AllStaticPresubmits` where in all other test files you're accessing the field directly. I think the way with the method is cleaner.

Comment (Ressetkk) [reply development/tools/jobs/testinfra/validate_test.go:16]: Got it changed in the latest commit.

**Pull request merged**